### PR TITLE
net: lwm2m_client_utils: Match tau with lifetime

### PR DESCRIPTION
**Pull request by juhaylinen**
Try to match periodic TAU with client lifetime.
Log TAU period and active time when PSM parameters are updated.

Update lwm2m_client sample's RPTAU config to match configured lifetime.

### CODE REVIEW

File: subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_lte_notification.c [hunk 1]
```diff
@@ -88,6 +91,16 @@ void lte_notify_handler(const struct lte_lc_evt *const evt)
...
 #endif /* CONFIG_LTE_LC_TAU_PRE_WARNING_NOTIFICATIONS */
+	case LTE_LC_EVT_PSM_UPDATE:
+		err = lte_lc_psm_get(&psm_cfg.tau, &psm_cfg.active_time);
+
+		if (err) {
+			LOG_WRN("Failed to get PSM information");
+			return;
+		}
+		LOG_INF("Tau period %d seconds", psm_cfg.tau);
+		LOG_INF("Active time %d seconds", psm_cfg.active_time);
```
Comment (SeppoTakalo) [inline subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_lte_notification.c:102]:
```suggestion
		LOG_DBG("Tau period %d seconds", psm_cfg.tau);
		LOG_DBG("Active time %d seconds", psm_cfg.active_time);
```

File: subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_utils_con_management.c [hunk 2]
```diff
@@ -57,6 +57,41 @@ static void lwm2m_utils_dtls_load(struct lwm2m_ctx *ctx)
...
+	}
+
+	if (lwm2m_get_u32(&LWM2M_OBJ(1, client->srv_obj_inst, 1), &lifetime) < 0) {
+		lifetime = CONFIG_LWM2M_ENGINE_DEFAULT_LIFETIME;
+		LOG_INF("Using default lifetime: %u", lifetime);
+	}
+
+	/* Try to match periodic TAU with lifetime*/
+	if (psm_cfg.tau != lifetime) {
+		LOG_INF("Requesting new periodic TAU");
```
Comment (SeppoTakalo) [inline subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_utils_con_management.c:85]:
```suggestion
		LOG_DBG("Requesting new periodic TAU");
```

File: subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_utils_con_management.c [hunk 3]
```diff
@@ -57,6 +57,41 @@ static void lwm2m_utils_dtls_load(struct lwm2m_ctx *ctx)
...
+	}
+
+	if (psm_cfg.active_time < 0) {
+		LOG_DBG("PSM deactivated");
+		return;
+	}
+
+	if (lwm2m_get_u32(&LWM2M_OBJ(1, client->srv_obj_inst, 1), &lifetime) < 0) {
+		lifetime = CONFIG_LWM2M_ENGINE_DEFAULT_LIFETIME;
+		LOG_INF("Using default lifetime: %u", lifetime);
```
Comment (SeppoTakalo) [inline subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_utils_con_management.c:80]:
```suggestion
		LOG_WRN("Using default lifetime: %u", lifetime);
```

I assume that this should not happen usually.

**Review by SeppoTakalo — APPROVED** [review]
Looks OK.

I would still change the debug level.

**Comment by SeppoTakalo** [comment]
Btw. This changes the behaviour of the client, so it deserves a changelog entry.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -445,7 +445,10 @@ Cellular samples
     * Release Assistance Indication (RAI) feature.
       This helps to save power by releasing the network connection faster on a network that supports it.
 
-  * Updated the eDRX cycle to 5.12s for both LTE-M and Nb-IOT.
+  * Updated:
+
+    * The eDRX cycle to 5.12s for both LTE-M and Nb-IOT.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:450]:
```suggestion
    * The eDRX cycle to 5.12 s for both LTE-M and NB-IoT.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 5]
```diff
@@ -866,7 +869,10 @@ Libraries for networking
 
 * :ref:`lib_lwm2m_client_utils` library:
 
-  * Updated the Release Assistance Indication (RAI) support to follow socket state changes from LwM2M engine, and modify RAI values based on the state.
+  * Updated:
+
+    * The Release Assistance Indication (RAI) support to follow socket state changes from LwM2M engine, and modify RAI values based on the state.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:874]:
```suggestion
    * The Release Assistance Indication (RAI) support to follow socket state changes from LwM2M engine and modify RAI values based on the state.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Comment by SeppoTakalo** [comment]
@juhaylinen Please check CI failures.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 6]
```diff
@@ -445,7 +445,10 @@ Cellular samples
     * Release Assistance Indication (RAI) feature.
       This helps to save power by releasing the network connection faster on a network that supports it.
 
-  * Updated the eDRX cycle to 5.12s for both LTE-M and Nb-IOT.
+  * Updated:
+
+    * The eDRX cycle to 5.12 s for both LTE-M and NB-IOT.
```
[the remark this answers is not in the document]
Comment (juhaylinen) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: Oops. Now fixed

**Pull request merged**